linker/shared/src/main/scala/org/scalajs/linker/frontend/LinkTimeProperties.scala

```diff
@@ -38,31 +39,29 @@ private[linker] final class LinkTimeProperties (
       LinkTimeString(ScalaJSVersions.current)
   )
 
-  def validate(name: String, tpe: jstpe.Type): Boolean = {
-    linkTimeProperties.get(name).exists {
-      case _: LinkTimeBoolean => tpe == jstpe.BooleanType
-      case _: LinkTimeInt     => tpe == jstpe.IntType
-      case _: LinkTimeString  => tpe == jstpe.StringType
-    }
-  }
+  def get(name: String): Option[LinkTimeValue] =
+    linkTimeProperties.get(name)
+}
 
-  def transformLinkTimeProperty(prop: js.LinkTimeProperty): js.Literal = {
-    val value = linkTimeProperties.getOrElse(prop.name,
-        throw new IllegalArgumentException(s"link time property not found: '${prop.name}' of type ${prop.tpe}"))
-    value match {
-      case LinkTimeBoolean(value) =>
-        js.BooleanLiteral(value)(prop.pos)
-      case LinkTimeInt(value) =>
-        js.IntLiteral(value)(prop.pos)
-      case LinkTimeString(value) =>
-        js.StringLiteral(value)(prop.pos)
+object LinkTimeProperties {
+  sealed abstract class LinkTimeValue {
+    def tpe: Type = this match {
+      case LinkTimeInt(_)     => IntType
+      case LinkTimeBoolean(_) => BooleanType
+      case LinkTimeString(_)  => StringType
```

Comment:
Nit: Instead of self matching, pass the type as val constructor param to `LinkTimeValue`?